Scimilarity metrics section to supplemental report

allyhawkins opened the pull request:
Closes #1031 

Here I'm adding a new section to the automated annotations QC for `SCimilarity` results. As noted in the [cell type annotation vignette](https://genentech.github.io/scimilarity/notebooks/cell_annotation_tutorial.html#Annotation-QC), we can use the `min_dist` metric as a confidence measure. 

I ended up using a similar plot to the `SingleR` plot, where the annotation is on the y-axis and the x-axis is the minimum distance metric. This illustrates the distribution of the minimum distance across all cells for each predicted cell type. I played around with UMAPs too, but ultimately the sina plot does a better job of displaying the distribution and gives the information we are looking for. 

As noted in the vignette, they use a value of 0.05 where any cell with a minimum distance from the query cells > 0.05 has less confidence in the prediction. I mentioned this in the description and then also added a dashed line for this value, which I think is helpful when we have lots of cell types and the plot gets really long. 

Are there other things that we should be sure to include in the text or point out about these results? 

While here, I also moved the `lump_wrap_celltypes()` function back to the `celltype-functions.rmd` file. I noticed in #1022, where it was removed, there was a comment that the reports were generated without the celltype sections, which is probably why it needed to be moved. In reality, `inferCNV` does not run without cell typing, so it did not need to be moved and addresses the concern brought up in https://github.com/AlexsLemonade/scpca-nf/pull/1049#pullrequestreview-3325710636.

Here's a copy of the supplemental report for three different libraries:
[reports.zip](https://github.com/user-attachments/files/22892167/reports.zip)

jashapiro commented on templates/qc_report/celltypes_supplemental_report.rmd:
```diff
@@ -85,12 +85,13 @@ has_clusters <- "cluster" %in% names(colData(processed_sce))
+  # keep consensus as the first method so it shows up first in comparison plots
+  ifelse(has_consensus, "Consensus", NA),
   ifelse(has_submitter, "Submitter", NA),
   ifelse(has_openscpca, "OpenScPCA", NA),
```
I was thinking just having the consensus before the three automated, so keeping the Submitter and OpenScPCA first: 
```suggestion
  ifelse(has_submitter, "Submitter", NA),
  ifelse(has_openscpca, "OpenScPCA", NA),
  ifelse(has_consensus, "Consensus", NA),
```

jashapiro replied:
I realized that this whole section can actually be simplified a bit by not using `ifelse`, but just regular `if` statements: 
```
# what celltypes are available?
available_celltypes <- c(
  if(has_submitter) "Submitter",
  if(has_openscpca) "OpenScPCA",
  if(has_consensus) "Consensus",
  if(has_singler) "SingleR",
  if(has_cellassign) "CellAssign",
  if(has_scimilarity) "SCimilarity"
)

# use an empty character vector if no celltypes present
if(is.null(available_celltypes){
  available_celltypes <- character()
}

```

This should save the `NA` removal and conversion, though you need to add a check that you don't end up with just `NULL`

allyhawkins commented:
Alright, I added back "other" and made sure they appear in the plots at the bottom of the list but before unknowns, which only applies to CellAssign. 

I also simplified the available celltypes chunk that you suggested.
